### PR TITLE
Add bitbox02

### PR DESCRIPTION
**Pull request by reardencode**
###### Motivation for this change

Preparatory work for supporting BitBox02 in electrum whenever electrum 4.0 comes out.

Removes python2 support from `block-io`, `streamlit`, and `zeronet`, hopefully we're not caring about losing python2 support these days?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6428,6 +6428,12 @@
     githubId = 1973389;
     name = "Reuben D'Netto";
   };
+  reardencode = {
+    email = "freedom@reardencode.com";
+    github = "reardencode";
+    githubId = 730881;
+    name = "Brandon Black";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:6436]: Please make this into a separate commit:

````
maintainers: add <name>
```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -6821,6 +6821,10 @@ in {
 
   ckcc-protocol = callPackage ../development/python-modules/ckcc-protocol { };
 
+  bitbox02 = callPackage ../development/python-modules/bitbox02 { };
+
+  noiseprotocol = callPackage ../development/python-modules/noiseprotocol { };
+
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:6827]: please attempt to sort these with the other packages, don't want to contribute more to the mess

**Review by jonringer — COMMENTED** [review]
each package change should be it's own commit, I see 2 package additions, 1 bump, 1 modification, and one maintainer addition. so there should roughly be 5 commits

To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Comment by nyanloutre** [comment]
@ofborg build python3Packages.block-io python3Packages.py-multihash zeronet streamlit

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (reardencode) [reply maintainers/maintainer-list.nix:6436]: done

**Comment by reardencode** [comment]
@nyanloutre I'm quite new to contributing to nixpkgs, is there a proper procedure for dropping python2 support from `streamlit`, `zeronet`, and `block-io`?  I don't think anything changed for us with `py-multihash`

Thanks!

File: pkgs/development/python-modules/bitbox02/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "bitbox02";
+  version = "2.0.3";
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1mcbg7bvlsii1kj2mm5686z01pwcc36qbqlhz4a8syjrzjm6pl2k";
+  };
+
+  doCheck = false;
```
Comment (prusnak) [inline pkgs/development/python-modules/bitbox02/default.nix:25]: Please add a comment why you skip tests or fix the tests, so they don't need to be skipped.

File: pkgs/development/python-modules/bitbox02/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [ hidapi noiseprotocol protobuf ecdsa semver typing-extensions base58 ];
+
+  meta = with stdenv.lib; {
+    description = "Communicate with your BitBox02 using Python";
+    homepage = https://github.com/digitalbitbox/bitbox02-firmware;
```
Comment (prusnak) [inline pkgs/development/python-modules/bitbox02/default.nix:31]: Use quotes for URLs (as per RFC45)

File: pkgs/development/python-modules/bitbox02/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{ stdenv
+, buildPythonPackage
+, hidapi
+, noiseprotocol
+, protobuf
+, ecdsa
+, semver
+, typing-extensions
+, base58
+, lib
```
Comment (prusnak) [inline pkgs/development/python-modules/bitbox02/default.nix:10]: Drop `lib`, it is not used anywhere.

File: pkgs/development/python-modules/noiseprotocol/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, buildPythonPackage
+, lib
```
Comment (prusnak) [inline pkgs/development/python-modules/noiseprotocol/default.nix:3]: Drop `lib`, it's not used anywhere.

File: pkgs/development/python-modules/noiseprotocol/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  checkPhase = ''
+    pytest
+  '';
+
+  buildInputs = [ scapy ];
+  propagatedBuildInputs = [ cryptography ];
+
+  meta = with stdenv.lib; {
+    description = "Python Implementation of Noise Protocol Framework";
+    homepage = https://github.com/plizonczyk/noiseprotocol;
```
Comment (prusnak) [inline pkgs/development/python-modules/noiseprotocol/default.nix:33]: Use quotes for URLs (as per RFC45)

File: pkgs/development/python-modules/bitbox02/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+  doCheck = false;
+
+  propagatedBuildInputs = [ hidapi noiseprotocol protobuf ecdsa semver typing-extensions base58 ];
+
+  meta = with stdenv.lib; {
+    description = "Communicate with your BitBox02 using Python";
+    homepage = https://github.com/digitalbitbox/bitbox02-firmware;
+    license = licenses.asl20;
+    maintainers = [ maintainers.reardencode ];
```
Comment (prusnak) [inline pkgs/development/python-modules/bitbox02/default.nix:33]:
```suggestion
    maintainers = with maintainers; [ reardencode ];
```

File: pkgs/development/python-modules/noiseprotocol/default.nix [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    license = licenses.mit;
+    maintainers = [ maintainers.reardencode ];
```
Comment (prusnak) [inline pkgs/development/python-modules/noiseprotocol/default.nix:35]:
```suggestion
    maintainers = with maintainers; [ reardencode ];
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by prusnak** [comment]
@reardencode Can you please rebase the PR, there are conflicts. While you are at it, please check whether some of the packages cannot be updated to even more recent versions. Finally, please prefix your commit messages with `python3Packages.`  so it reads `python3Packages.base58` instead of `base58`. Thanks!

**Review by prusnak — CHANGES_REQUESTED** [review]
@reardencode Can you please rebase the PR, there are conflicts. While you are at it, please check whether some of the packages cannot be updated to even more recent versions. Finally, please prefix your commit messages with `python3Packages.`  so it reads `python3Packages.base58` instead of `base58`. Thanks!
